server/src/main/java/org/opensearch/action/admin/cluster/snapshots/create/TransportCreateSnapshotAction.java

```diff
@@ -103,7 +111,9 @@ protected void clusterManagerOperation(
         ClusterState state,
         final ActionListener<CreateSnapshotResponse> listener
     ) {
-        if (request.waitForCompletion()) {
+        Repository repository = repositoriesService.repository(request.repository());
+        boolean isSnapshotV2 = SNAPSHOT_V2.get(repository.getMetadata().settings());
```

Comment:
>What's the reason that we need to have sync behaviour in create snapshot?

Since the snapshot operation is executed entirely on the cluster manager node with snapshot v2 and the execution is pretty faster (unrelated to shard count) with pinned timestamp, we can execute the complete operation in a very short time(few seconds to a few minutes). Sync execution eliminates the need for state management and makes snapshot lighter

Reply:
>Do we need to handle BWC where not all the nodes have moved to the latest version yet?
I don't think so, since the snapshot operation is executed completely in the cluster manager node.

Reply:
> > Do we need to handle BWC where not all the nodes have moved to the latest version yet?
> 
> I don't think so, since the snapshot operation is executed completely in the cluster manager node.

What will happen lets say where a data node is on the older OS version and deletes the segments/translog corresponding to the pinned timestamp since it does not understands the ultralight snapshots.

Reply:
added min version check for backward compatibility